src/ynEIGEN/AssetRegistry.sol

```diff
@@ -0,0 +1,342 @@
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bytes32 public constant UNPAUSER_ROLE = keccak256("UNPAUSER_ROLE");
+    
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  VARIABLES  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+     /// @notice List of supported ERC20 asset contracts.
+    IERC20[] public assets;
+
+    mapping(IERC20 => AssetData) public _assetData;
+
+    bool public actionsPaused;
+
+    IRateProvider public rateProvider;
+    IEigenStrategyManager public eigenStrategyManager;
+    IynEigen ynEigen;
+
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  INITIALIZATION  ----------------------------------
+    //--------------------------------------------------------------------------------------
+
+    constructor() {
+       _disableInitializers();
+    }
+
+    struct Init {
+        IERC20[] assets;
+        IRateProvider rateProvider;
+        IEigenStrategyManager eigenStrategyManager;
+        IynEigen ynEigen;
+        address admin;
+        address pauser;
+        address unpauser;
+    }
+
+    function initialize(Init calldata init)
+        public
+        notZeroAddress(address(init.rateProvider))
+        notZeroAddress(address(init.admin))
+        notZeroAddress(address(init.pauser))
+        notZeroAddress(address(init.unpauser))
+        initializer {
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.pauser);
+        _grantRole(UNPAUSER_ROLE, init.unpauser);
+
+        uint256 assetsLength = init.assets.length;
+        for (uint256 i = 0; i < assetsLength; i++) {
+            if (address(init.assets[i]) == address(0)) {
+                revert ZeroAddress();
+            }
+            assets.push(init.assets[i]);
+            _assetData[init.assets[i]] = AssetData({
+                active: true
+            });
+        }
+
+        rateProvider = init.rateProvider;
+        eigenStrategyManager = init.eigenStrategyManager;
+        ynEigen = init.ynEigen;
+    }
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ASSET MANAGEMENT  --------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Adds a new asset to the system.
+     * @dev Adds an asset to the _assetData mapping and sets it as active. This function can only be called by the strategy manager.
+     * @param asset The address of the ERC20 token to be added.
+     */
+    function addAsset(IERC20 asset)
```

Comment:
If you disable and asset using `disableAsset` you can then call `addAsset` and it will add it again, creating a duplicate entry in the `assets` array. And this will lead to the same problem with `deleteAsset` as described [here](https://github.com/yieldnest/yieldnest-protocol/pull/121/files#r1679042243)